analyses/mutational-signatures/04-analyze_de_novo.Rmd

```diff
@@ -0,0 +1,175 @@
+---
+title: "De novo mutational signature extraction from WGS data"
+author: "S. Spielman"
+date: "2021"
+output:
+  html_notebook:
+    toc: yes
+    toc_float: yes
+  html_document:
+    toc: yes
+    df_print: paged
+editor_options:
+  chunk_output_type: inline
+---
+
+
+<br><br><br><br>
+
+#### Packages and paths
+
+
+```{r}
+# Load libraries
+library(sigfit)
+`%>%` <- dplyr::`%>%`
+
+# Path to input data
+path_to_input_rds <- file.path("..", "..", "scratch", "mutational-signatures", "extraction")
+
+# Result path
+path_to_results <- "results/"
+if (!dir.exists(path_to_results)) {
+  dir.create(path_to_results, recursive = TRUE)
+}
+
+# De novo signatures and exposures list (condensed from sigfit output) file
+de_novo_signatures_file <- file.path(path_to_results, "de_novo_signatures.RDS")
+de_novo_exposures_file <- file.path(path_to_results, "de_novo_exposures.RDS")
+
+# Load cosmic data from sigfit
+data("cosmic_signatures_v3")
+cosmic_names <- row.names(cosmic_signatures_v3)
+```
+
+<br><br>
+First, we need to extract the signatures and exposures from the *de novo* extraction RDS files. We save each into a list `de_novo_signatures` and `de_novo_extractions` separately, and write each list to file. If _both_ files already exist, we can simply read it in the lists.
+
+```{r parse_signatures}
+if (file.exists(de_novo_signatures_file) & file.exists(de_novo_exposures_file))
+{
+  de_novo_signatures <- readr::read_rds(de_novo_signatures_file)
+  de_novo_exposures <- readr::read_rds(de_novo_exposures_file)
+
+} else {
+    
+  # Collect the de novo extracted signatures into a single list
+  # This takes a few minutes to parse.
+  de_novo_signatures <- list()
+  de_novo_exposures <- list()
+  for (this_model in c("poisson" , "multinomial"))
+  {
+    for (this_seed in 1:5){
+  
+      this_name <- glue::glue("seed_{this_seed}_model_{this_model}")
+      filename <- file.path(path_to_input_rds, 
+                            glue::glue("{this_name}.RDS"))
+      #cat(filename)
+      if (file.exists(filename)) {
+        fit <- readr::read_rds(filename)
+        for (k in 2:5){
+          sig_name <- glue::glue("{k}_{this_name}")
+          de_novo_signatures[[ sig_name ]] <- sigfit::retrieve_pars(fit[[k]], par = "signatures") 
+          de_novo_exposures[[ sig_name ]] <- sigfit::retrieve_pars(fit[[k]], par = "exposures") 
+        }
+      }
+    }
+    
+  }
+  # We save this list to a result file
+  readr::write_rds(de_novo_signatures, de_novo_signatures_file)
+  readr::write_rds(de_novo_exposures, de_novo_exposures_file)
+}
+```
+
+<br><br>
+Based on goodness of fit analyses, we expect these *k* to be optimal for the given seed and inference model (noting that we do see strong sensitivity to this starting condition).
```

Comment:
If I'm coming here for the first time, where would I look to see if I agree with your interpretation? 😄

Reply:
Missed this comment..Are you implying "add citations to Rmd"?

Reply:
Maybe? If I recall correctly, the analysis to support this statement isn't in this notebook – it's upstream. Where do I find the plots, etc. in this repo?

Reply:
Ah, I see what you mean. I thought you were asking for more details about "but why use cosine similarity." An explanation like "where are the plots for your magic numbers" is a good addition, clearly.